### PR TITLE
'shift' moves between monitors if appropriate

### LINKED ISSUE #1066: shift command should move between monitors
The `shift` command should move windows to a neighboured monitor if it reaches the monitor boundary. Have configured it manually https://github.com/t-wissmann/dotfiles/blob/master/config/herbstluftwm/autostart#L208..L233 but I think it should be the default behaviour (just like it is already for `focus`)

### PR DESCRIPTION
**Pull request by t-wissmann**
If 'shift' cannot move a window within a tag, then it moves the window
to the next monitor in the specified direction.

Test cases for hopefully all scenarios are included in this change.

This fixes #1066.

### CODE REVIEW

File: tests/test_layout.py [hunk 1]
```diff
@@ -1377,6 +1377,99 @@ def test_shift_no_neighbour_frame(hlwm, split):
...
+
+@pytest.mark.parametrize("floating", [True, False])
+def test_shift_stays_on_monitor(hlwm, floating):
+    hlwm.call('add othertag')
+    # monitor 1 below monitor 0
+    hlwm.call('set_monitors 800x600+0+0 800x600+0+800')
+    if floating:
+        hlwm.attr.tags.focus.floating = 'on'
+    else:
+        # create empty from at the bottom
```
Comment (blarz) [inline tests/test_layout.py:1439]: frame?

File: tests/test_layout.py [hunk 2]
```diff
@@ -1377,6 +1377,99 @@ def test_shift_no_neighbour_frame(hlwm, split):
...
+        hlwm.call('split bottom')
+
+    winid, _ = hlwm.create_client(position=(0, 0))
+    assert hlwm.attr.clients.focus.winid() == winid
+    assert hlwm.attr.monitors.focus.index() == '0'
+
+    # the new client is far away from the bottom edge of monitor 0.
+    # so shifting it downwards make it stay on monitor 0
```
Comment (blarz) [inline tests/test_layout.py:1447]: makes?

**Review by blarz — COMMENTED** [review]
Very nice! Also works locally. Just some typos in test_ comments

File: tests/test_layout.py [hunk 3]
```diff
@@ -1377,6 +1377,99 @@ def test_shift_no_neighbour_frame(hlwm, split):
...
+@pytest.mark.parametrize("cross_monitor_bounds", [True, False])
+def test_shift_to_other_monitor_if_allowed_by_setting(hlwm, cross_monitor_bounds):
+    hlwm.attr.settings.focus_crosses_monitor_boundaries = hlwm.bool(cross_monitor_bounds)
+    hlwm.call('add othertag')
+    # monitor 1 right of monitor 0
+    hlwm.call('set_monitors 800x600+0+0 800x600+800+0')
+
+    hlwm.call('rule focus=on switchtag=off')
+    winid, _ = hlwm.create_client()
+    # put another windo on the other tag
```
[the remark this answers is not in the document]
Comment (t-wissmann) [reply tests/test_layout.py]: Thanks! I indeed forgot to put another client on the other tag. (This ensures that the shifted client properly keeps the focus).

File: tests/test_layout.py (hunk 1 above)
Comment (t-wissmann) [reply tests/test_layout.py]: Yes :) fixed.

File: tests/test_layout.py (hunk 2 above)
Comment (t-wissmann) [reply tests/test_layout.py]: Yes, fixed :-)

Comment (blarz) [reply tests/test_layout.py]: Thanks!

File: tests/test_layout.py (hunk 1 above)
Comment (blarz) [reply tests/test_layout.py]: Thanks!

File: tests/test_layout.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (blarz) [reply tests/test_layout.py]: Cool, thanks for fixing and adding this :)

**Review by blarz — APPROVED** [review]
Cool!

**Pull request merged**